hasura/graphql-engine
# allow altering type of a column whose permissions defined only with session variables (close #2070)

### Description



Allow altering type of a column even if it is being used in permissions with session variable values.

### Affected components 


- Server
- Tests

### Related Issues



close #2070 

### Solution and Design


- Add column dependencies with reason `sess_var` if its permissions are defined only with session variables
- In `run_sql` query flow, if column types are altered and `sess_var` dependant permissions are found then rebuild those permission infos and update the schema cache.

### Steps to test and verify


- Create any permission having a column with session variable
- Try to alter the type of that column, it should go through.

### Limitations, known bugs & workarounds
